canonical-web-and-design/ubuntu.com
# Corrects alt= text across the site

## Done

Corrects `alt=` text across the site for improved accessibility.

The problems fall into four main categories:
- Logorrhea: for example, our hyperscale partners are not Hewlett Packard Enterprise logo, Arm logo, and Cavium logo — they are Hewlett Packard Enterprise, Arm, and Cavium. (It’s hardly ever appropriate to include the word “logo” in `alt=` text.)
- Redundancy: using `alt=` simply to name an organisation/product (with or without “logo”) when an adjacent heading already does that.
- Mystery: for example, `alt="header image"`, `alt="image for {{ header_title }}"`, or `alt="OS image size comparison"`.
- Typos: for example, “Bloomburg”, “Deutshe Telekom”, and, for AT&T, “spiculelogo”.

 In correcting them I’ve followed [the approach prescribed by the HTML spec](https://html.spec.whatwg.org/multipage/images.html#alt), specifically:

- For [a link or button containing nothing but the image](https://html.spec.whatwg.org/multipage/images.html#a-link-or-button-containing-nothing-but-the-image), I’ve retained the name of the entity, for example, `<a href="/"><img src="`…`logo-ubuntu-white.svg" alt="Ubuntu logo white"></a>` becomes `<a href="/"><img src="`…`logo-ubuntu-white.svg" alt="Ubuntu"></a>`. (There are some cases where an image-only link is adjacent to a text-only link to the same place. Ideally these would be a single link containing both, but fixing those would be much trickier.)

- For [a phrase or paragraph with an alternative graphical representation](https://html.spec.whatwg.org/multipage/images.html#a-phrase-or-paragraph-with-an-alternative-graphical-representation:-charts,-diagrams,-graphs,-maps,-illustrations) I’ve provided equivalent text: `alt="Ubuntu Core: 260 MB. Red Hat coreOS: 807 MB."` and `alt="Including AT&amp;T, eBay, NTT, Bloomberg, Cisco, Deutsche Telekom, Walmart and Best Buy."`

- In remaining cases where a “[logo is being used next to the name of the entity that it represents, then the logo is supplemental, and its `alt` attribute must instead be empty](https://html.spec.whatwg.org/multipage/images.html#a-short-phrase-or-label-with-an-alternative-graphical-representation:-icons,-logos)”. (For review, in many cases it will be necessary to expand the diff to check that the name of the entity is repeated prominently nearby.)

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- Visit the site with a text-only browser or screenreader at: http://0.0.0.0:8001/

## Issue / Card

None
